storage/netapp_e/na_eseries_facts.py

```diff
@@ -0,0 +1,194 @@
+        api_url: "{{ netapp_api_url }}"
+        api_username: "{{ netapp_api_username }}"
+        api_password: "{{ netapp_api_password }}"
+        validate_certs: "{{ netapp_api_validate_certs }}"
+"""
+
+RETURN = """
+msg: Gathered facts for <StorageArrayId>.
+"""
+
+
+def request(url, data=None, headers=None, method='GET', use_proxy=True,
+            force=False, last_mod_time=None, timeout=10, validate_certs=True,
+            url_username=None, url_password=None, http_agent=None, force_basic_auth=True, ignore_errors=False):
+    try:
+        r = open_url(url=url, data=data, headers=headers, method=method, use_proxy=use_proxy,
+                     force=force, last_mod_time=last_mod_time, timeout=timeout, validate_certs=validate_certs,
+                     url_username=url_username, url_password=url_password, http_agent=http_agent,
+                     force_basic_auth=force_basic_auth)
+    except HTTPError:
+        err = get_exception()
+        r = err.fp
+
+    try:
+        raw_data = r.read()
+        if raw_data:
+            data = json.loads(raw_data)
+        else:
+            data = None
+    except:
+        if ignore_errors:
+            pass
+        else:
+            raise
+
+    resp_code = r.getcode()
+
+    if resp_code >= 400 and not ignore_errors:
+        raise Exception(resp_code, data)
+    else:
+        return resp_code, data
+
+
+def main():
+    argument_spec = basic_auth_argument_spec()
+    argument_spec.update(
+        ssid=dict(required=True))
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    p = module.params
+
+    ssid = p['ssid']
+    validate_certs = p['validate_certs']
+
+    try:
+        api_usr = p['api_username']
+        api_pwd = p['api_password']
+        api_url = p['api_url']
+    except KeyError:
+        import os
+        api_usr = os.environ['API_USERNAME']
+        api_pwd = os.environ['API_PASSWORD']
+        api_url = os.environ['API_URL']
+
+    if not api_usr or not api_pwd or not api_url:
+        module.fail_json(msg="You must pass in api_username "
+                             "and api_password and api_url to the module. Alternatively "
+                             "you can specify API_URL, API_USERNAME and API_PASSWORD "
+                             "as environment variables.")
+    facts = dict(ssid=ssid)
+    result = dict(ansible_facts=facts, changed=False)
+
+    # fetch the list of storage-pool objects and look for one with a matching name
+    try:
+        (rc, resp) = request(api_url + "/storage-systems/{0}/graph".format(ssid),
+                             headers=dict(Accept="application/json"),
+                             url_username=api_usr, url_password=api_pwd, validate_certs=validate_certs)
+    except:
+        error = get_exception()
+        module.fail_json(
+            msg="Failed to obtain facts from storage array with id [{0}]. Error [{1}]".format(ssid, str(error)))
```

Comment:
.format() is only available in python2.6+.  Probably should use percent formatting for python-2.4 compatibility.

```
msg="Failed to obtain facts from storage array with id [%s]. Error [%s]" % ssid, str(error))
```